plugin/src/software/aws/toolkits/eclipse/amazonq/util/UpdateUtils.java

```diff
@@ -0,0 +1,121 @@
+        return INSTANCE;
+    }
+
+    private UpdateUtils() {
+        mostRecentNotificationVersion = Activator.getPluginStore().getObject(Constants.DO_NOT_SHOW_UPDATE_KEY, Version.class);
+        String localString = PluginClientMetadata.getInstance().getPluginVersion();
+        localVersion = ArtifactUtils.parseVersion(localString.substring(0, localString.lastIndexOf(".")));
+    }
+
+    private boolean newUpdateAvailable() {
+        //fetch artifact file containing version info from repo
+        remoteVersion = fetchRemoteArtifactVersion(REQUESTURL);
+
+        //return early if either version is unavailable
+        if (remoteVersion == null || localVersion == null) {
+            return false;
+        }
+
+        //prompt should show if never previously displayed or remote version is greater
+        boolean shouldShowNotification = mostRecentNotificationVersion == null || remoteVersionIsGreater(remoteVersion, mostRecentNotificationVersion);
+
+        return remoteVersionIsGreater(remoteVersion, localVersion) && shouldShowNotification;
+    }
+
+    public void checkForUpdate() {
+        if (newUpdateAvailable()) {
+            showNotification();
+        }
+    }
+
+    private Version fetchRemoteArtifactVersion(final String repositoryUrl) {
+        HttpURLConnection connection = null;
+        try {
+            URL url = new URL(repositoryUrl);
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setConnectTimeout(5000);
+            connection.setReadTimeout(5000);
+
+            // handle response codes
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                throw new AmazonQPluginException("HTTP request failed with response code: " + responseCode);
+            }
+
+            // process XZ content from input stream
+            try (InputStream inputStream = connection.getInputStream();
+                 XZInputStream xzis = new XZInputStream(inputStream);
+                 BufferedReader reader = new BufferedReader(new InputStreamReader(xzis))) {
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (line.contains("<artifact classifier=\"osgi.bundle\"")) {
+                        int versionStart = line.indexOf("version=\"") + 9;
+                        int versionEnd = line.indexOf("\"", versionStart);
+                        String fullVersion = line.substring(versionStart, versionEnd);
+                        return ArtifactUtils.parseVersion(fullVersion.substring(0, fullVersion.lastIndexOf(".")));
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+        return null;
```

Comment:
We should avoid using a raw `HttpUrlConnection` which effectively puts us on the hook to handle errors and retries. Try and grab the client instance from the `HttpClientFactory` and use that instead.